homeassistant/helpers/entity_component.py

```diff
@@ -154,6 +160,8 @@ def _async_setup_platform(self, platform_type, platform_config,
         except Exception:  # pylint: disable=broad-except
             self.logger.exception(
                 'Error while setting up platform %s', platform_type)
+        finally:
+            warn_task.cancel()
```

Comment:
Why not just check the timer here instead of scheduling a task? (Except the case where a setup is stuck forever)